### DISCUSSION TITLE
How can I use best model to process SHAP util?

### DISCUSSION THREAD

**Discussion by Tonywhitemin** [mljar/mljar-supervised#599] [Q&A]
Hi @pplonski :

First, wish you have good new year holidays!
As title, I'd like to use the best model after " fit()" to predict the test dataset I prepared and use SHAP utils.
The code as below:

```py
test_df=pd.read_csv("/home/tony/use_case_data/testdata.csv")

X_test=test_df.drop("label",axis=1)
y_test=test_df["label"]

automl = AutoML(results_path=r"/home/tony/文件/AutoML01") 


explainer = shap.TreeExplainer(automl)


shap_values = explainer.shap_values(X_test)

shap.plots.bar(shap_values) 
```

But I've got the exception message as below:
```
**"shap.utils._exceptions.InvalidModelError: Model type not yet supported by TreeExplainer: <class 'supervised.automl.AutoML'>"**
```
The best model is "4_Default_Xgboost" and I think I can use "shap.TreeExplainer" as tool...
Could you tell me how to solve this situation?
Thank!

**Comment by pplonski** [comment]
The AutoML object is not a single model, and `shap` needs to accept a single model. You can use AutoML to search for best parameters and best preprocessing and then manually train a single custom model to use shap on it.

**Reply by Tonywhitemin** [reply]
Thanks for your quick reply!
